packages/react-component-library/src/components/CheckboxRadioBase/CheckboxRadioBase.tsx

```diff
@@ -36,7 +37,13 @@ export const CheckboxRadioBase = React.forwardRef<
   ) => {
     const id = useExternalId(`${type}-input`, externalId)
     const localRef = useRef<HTMLInputElement>(null)
-    const [isChecked, setIsChecked] = useState(defaultChecked)
+    const [isChecked, setIsChecked] = useState<boolean>(
+      defaultChecked || checked
+    )
+
+    useEffect(() => {
+      setIsChecked(checked)
+    }, [checked])
```

Comment:
If `defaultChecked` is used then the background ends up in the wrong state which seems to be due to the useEffect running when the component is mounted:

![Screencast_04-04-22_11:50:39](https://user-images.githubusercontent.com/66470099/161529352-676fa8c6-3a42-40d1-bad5-9530a14e06ec.gif)

So I think this useEffect would need a guard in the body to not do anything in that case.

Alternatively, another approach may be to rejig things to try to do this (should get rid of the need for a useEffect I think):

- if the component is controlled, just use `checked` (and don't worry about maintaining `isChecked`)
- if the component is uncontrolled, then try to passively monitor the state and maintain `isChecked`
- (make the stories use defaultChecked again)